internal/oci/runtime_oci.go

```diff
@@ -1001,7 +1001,12 @@ func (r *runtimeOCI) UpdateContainerStatus(ctx context.Context, c *Container) er
 		// Collect metric by container name
 		metrics.Instance().MetricContainersOOMCountTotalInc(c.Name())
 	}
-
+	// If this container had a node level PID namespace, then any children processes will be leaked to init.
+	// Eventually, the processes will get cleaned up when the pod cgroup is cleaned by the kubelet,
+	// but this situation is atypical and should be avoided.
+	if c.nodeLevelPIDNamespace() {
```

Comment:
Is there a race possible where this could end up getting called twice? Is the call re-entrant?
Another approach may be to add a call to signal in the exit monitor.

Reply:
since this call happens after we set the ExitCode for the container, there's no way to get back here (as we stop above if ExitCode has been defined)